electrum/trampoline.py

```diff
@@ -185,17 +131,47 @@ def _extend_trampoline_route(
         start_node = route[-1].end_node
     trampoline_features = LnFeatures.VAR_ONION_OPT
     # get policy for *start_node*
-    policy = trampoline_policy(trampoline_fee_level)
+    # note: trampoline nodes are supposed to advertise their fee and cltv in node_update message.
+    #       However, in the temporary spec, they do not.
+    #       They also don't send their fee policy in the error message if we lowball the fee...
     route.append(
         TrampolineEdge(
             start_node=start_node,
             end_node=end_node,
-            fee_base_msat=policy['fee_base_msat'] if pay_fees else 0,
-            fee_proportional_millionths=policy['fee_proportional_millionths'] if pay_fees else 0,
-            cltv_delta=policy['cltv_expiry_delta'] if pay_fees else 0,
+            fee_base_msat=PLACEHOLDER_FEE if pay_fees else 0,
+            fee_proportional_millionths=PLACEHOLDER_FEE if pay_fees else 0,
+            cltv_delta=576 if pay_fees else 0,
             node_features=trampoline_features))
 
 
+def _allocate_fee_along_route(
+    route: List[TrampolineEdge],
+    *,
+    budget: PaymentFeeBudget,
+    trampoline_fee_level: int,
+) -> None:
+    # calculate budget_to_use, based on given max available "budget"
+    if trampoline_fee_level == 0:
+        budget_to_use = 0
+    else:
+        assert trampoline_fee_level > 0
+        MAX_LEVEL = 6
+        if trampoline_fee_level > MAX_LEVEL:
+            raise NoPathFound()
+        budget_to_use = budget.fee_msat // (2 ** (MAX_LEVEL - trampoline_fee_level))
```

Comment:
Note: this logic means that the lowest total fee we will try is the user configured max divided by 32. E.g. by default the max is 1% of payment amount, i.e. `fee=1/3200*payment_amount`.

If the user configures a higher max budget, that raises the corresponding the lowest possible fee we will try.